proposal/calc-constants.md

```diff
@@ -0,0 +1,202 @@
+> not interpolated.
+
+## Semantics
+
+### `CalcValue`
+
+Add the following to the existing semantics for `CalcValue`:
+
+* If `value` is case-insensitively equal to `pi`, return 3.141592653589793.
+
+  > This is the closest double approximation of the mathematical constant π.
+
+* If `value` is case-insensitively equal to `e`, return 2.718281828459045.
+
+  > This is the closest double approximation of the mathematical constant e.
+
+* If `value` is case-insensitively equal to `infinity`, return the double
+  `Infinity`.
+
+* If `value` is case-insensitively equal to `-infinity`, return the double
+  `-Infinity`.
+
+* If `value` is case-insensitively equal to `nan`, return the double `NaN`.
+
+* If `value` is any other `<identifier>`, return a `CalcuationRaw` with `value`.
+
+## Serialization
+
+### Calculations
+
+#### `Number`
+
+To serialize a `Number` within a `CalculationExpression`:
+
+* If the number is [degenerate], [convert it to a calculation], then serialize
+  the resulting calculation's sole argument.
+
+  [degenerate]: #degenerate-number
+  [convert it to a calculation]: #converting-a-number-to-a-calculation
+
+* Otherwise, serialize the number as normal.
+
+#### `CalculationOperation`
+
+Add another possible condition for parenthesizing the right value:
+
+* the operator is `"+"`, `"-"`, or `"/"` and the right value is a degenerate
+  number with one or more units.
+
+### Numbers
+
+To serialize a number to CSS:
+
+* If the number has more than one numerator unit, or more than zero denominator
+  units, throw an error.
+
+* If the number is degenerate, [convert it to a calculation] then serialize that
+  to CSS.
+
+* Otherwise:
+
+  * Emit a string that can be parsed as a [`<number-token>`] with the
+    same value as the number.
+
+    [`<number-token>`]: https://www.w3.org/TR/css-syntax-3/#typedef-number-token
+
+  * If the number has a numerator unit, emit that unit.
+
+## Procedures
+
+### Converting a Number to a Calculation
+
+Given a number `number`, this procedure returns a CSS-compatible calculation
+that represents the same numeric value.
+
+* If `number` is `Infinity`, let `value` be a `CalculationRaw` whose `value` is
+  `'infinity'`.
+
+* Otherwise, if `number` is `-Infinity`, let `value` be a `CalculationRaw` whose
+  `value` is `'-infinity'`.
+
+* Otherwise, if `number` is `NaN`, let `value` be a `CalculationRaw` whose
+  `value` is `'NaN`.
+
+* Otherwise, let `value` be a `CalculationValue` whose value is `number` without
+  units.
+
+* For each unit `unit` in `number`'s numerator units:
+
+  * Set `value` to a `CalculationOperation` with `operator` set to `'*'`, `left`
+    set to `value`, and `right` set to a number with value 1 and unit `unit`.
+
+* For each unit `unit` in `number`'s denominator units:
+
+  * Set `value` to a `CalculationOperation` with `operator` set to `'/'`, `left`
+    set to `value`, and `right` set to a number with value 1 and unit `unit`.
```

Comment:
I imagine this procedure is here only for reference because the section above about numbers says an error would be thrown. Is there any other case that would trigger the "convert a number to a calculation" besides the number being degenerate?

Reply:
Not currently, but we'll probably need to do so later on once browsers start supporting complex units in calculations.